source/production/F0.Minesweeper.Logic.Abstractions/MinefieldGenerationOptions.cs

```diff
@@ -0,0 +1,10 @@
+namespace F0.Minesweeper.Logic.Abstractions
+{
+	public enum MinefieldGenerationOptions
+	{
+		Random,
+		Safe,
+		FirstEmpty,
```

Comment:
Either we make this enum flagged or `FirstEmpty` is part of something else. Because this value defines **when** the field is generated and the others **how** it is generated

Reply:
IMO the naming could be more descriptive, e.g. `Random`: there is always some kind of random invovled.
IIUC this represents the behavior on the first uncover: whether the first may be a mine, may not be a mine, may not have any neighboring mines or is always a mine.

Reply:
As discussed, I will change the names as follows:
* Enum => MinefieldFirstUnconverBehavior
* Random => MayYieldMine
* Safe => CannotYieldMine
* FirstEmpty => WithoutAdjacentMines
* Impossible => AlwaysYieldsMine